ag_data/models.py

```diff
@@ -49,10 +49,22 @@ class AGSensor(models.Model):
     """Stores the information about sensors including name and type id.
     """
 
-    id = models.AutoField(primary_key=True)
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    serial = models.IntegerField(unique=True, default=1)
```

Comment:
I think using uuid as a primary key will resolve the issue.
```
duplicate key value violates unique constraint "mercury_temperaturesensor_pkey"
DETAIL:  Key (id)=(2) already exists.
```

And  I'm also thinking that adding `serial` field might cause the same error because serifl field is unique. Can you confirm if the error doesn't happen with this field?

Reply:
Adding a non-pk that requires unique will likely to cause other issue during syncing. But that is part of another issue. This pr assumes that the other issue is working.

Reply:
Fair enough. 

My concern is that the `serial` will work similar as an auto increment field. The difference is that we write the auto increment feature by our own. And even the original issue #181 stated the issue on the primary key, however the actual issue was that it broke "unique constraint". 
```
duplicate key value violates unique constraint
```

Even though `serial` is non-pk, it's still unique. The problem might still exist with the similar reproduce steps and I'd like to check if this persists.

Reply:
sure, please verify this. the idea here is that the normal procedure of creating a sensor in the UI will be fine. And for possible duplicates in the syncing, the syncing mechanism will handle this.

Reply:
@alldne I added tests in the latest commit to check the behavior. 

The latest behavior will be like this: the project code will avoid the conflicting sensor serial if that happens. So the user will never see the error page when they create a new sensor.

On the other hand, our synchronization rules will further handle the situation when the user creates two sensors on two computers but they bear the same serial number. Now that is more relevant to the situation in #182.

Reply:
Sounds good to me